### DISCUSSION TITLE
Publish hashes of signing keys

### DISCUSSION THREAD

**Discussion by grrrrr** [mullvad/mullvadvpn-app#8941] [Feature requests & Ideas]
### I have checked if others have suggested this already

- [X] I have checked this issue tracker to see if others have reported similar issues.

### Feature description

On Android,  you can use [AppVerifier](https://github.com/soupslurpr/AppVerifier) to confirm if an apk was signed by the owners or an untrusted key.   This can be combined with Obtanium ( #6560 ) to check at install time.

The hashes could be published in a number of places for additional trust.  e.g

- blog post
- gpg signed download
- mastodon / twitter /...
- Github

### Alternative solutions

While cumbersome, continue installing apk from the website and verifying with gpg before moving apk to phone for install

### Type of feature

- [X] Better privacy/anonymity
- [ ] Better at circumventing censorship
- [X] Easier to use
- [ ] Other

### Operating System

- [X] Android
- [ ] iOS
- [ ] Windows
- [ ] macOS
- [ ] Linux

**Comment by albin-mullvad** [comment]
Thank you for the suggestion, we've created an internal issue to look into this.

**Comment by sad6auser** [comment]
Hi

The alternative solution you suggest is not only cumbersome but also has the expectation that you trust the desktop you are using to verify with GPG. For many people their desktop is a more insecure device than their phone. This only increases the need for the publication of Android signing certificate hashes imo that can be verified using Android apps like AppVerifier.

Ideally the hash should just be put on this overview https://mullvad.net/en/download/vpn/android 
If users get their app from GitHub releases they can verify it with the hash on the site. Both are in that case on a different domain, otherwise it would be kind of pointless because a compromise of GitHub could lead to both a replacement of the APK and of the published hash with tampered ones. Even better would indeed be to publish it in multiple places.

**Comment by harpogo** [comment]
> Even better would indeed be to publish it in multiple places.

Would propose to publish the hash at mullvad.net as @sad6auser mentioned as well as here on Github for example, so you can double check it. That would be an advantage because the apk download is not only available here on Github but on mullvad.net as well. And it's harder to compromise both, mullvad.net and the Github account at the same time than just one of them.
